### DISCUSSION TITLE
Is it possible to have the extension activated after LSP being activated?

### DISCUSSION THREAD

**Discussion by Vivecccccc** [microsoft/vscode-discussions#1207] [Extension Development QnA] [answered]
Hi, my extension needs to call `vscode.executeDocumentSymbolProvider` command for certain documents upon the extension being activated. However, I discovered that this command would not return anything other than `undefined` until syntax highlighting on the UI was effected.

Given that typescript features may be quickly activated, other languages may not be this case. My initial thought was to have my extension being activated *after* the LSP extension, e.g., `ms-python.python`, but this #595 proved it to be not possible. Still, I am expecting if extension of LSP may have any exempt.

Sincerely appreciate for any advice.

**Comment by tamuratak** [comment, marked as answer]
You can call `await sleep` to wait for LSP to be activated. While this might not be the ideal solution, it is straightforward.

- https://stackoverflow.com/a/39914235

**Reply by Vivecccccc** [reply]
Thank you. It seems to be the only possible workaround so far.
